tests/camel-blueprint-cxf-test/src/test/java/org/apache/camel/test/cxf/blueprint/CxfRsEndpointBeansTest.java

```diff
@@ -43,7 +57,40 @@ public void testCxfBusInjection() {
         JAXRSClientFactoryBean client = serviceEndpoint.createJAXRSClientFactoryBean();
         assertEquals("These cxfrs endpoints don't share the same bus", server.getBus().getId(), client.getBus().getId());
     }
-    
 
+    @Test
+    public void testDestinationOverrideURLHandling() {
+
+        try {
+            context.getRouteController().startRoute("url-override-route");
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        List<String> expected = Arrays.asList(
+                                              "foo1",
+                                              "foo2",
+                                              "foo1",
+                                              "foo2",
+                                              "foo1");
+
+        expected.forEach(host -> pT.send(exchange -> {
+            Message in = exchange.getIn();
+            in.setHeader(CxfConstants.CAMEL_CXF_RS_USING_HTTP_API, false);
+            in.setHeader(CxfConstants.OPERATION_NAME, "getCustomer");
+            in.setBody("Scott");
+            in.setHeader(Exchange.ACCEPT_CONTENT_TYPE, "application/json");
+            in.setHeader(Exchange.DESTINATION_OVERRIDE_URL, "http://" + host);
+            in.setHeader(Exchange.HTTP_METHOD, "GET");
+        }));
+
+        MockEndpoint mockEndpoint = getMockEndpoint("mock:resultURLOverride");
+        Assert.assertArrayEquals(expected.toArray(),
+                                 mockEndpoint.getExchanges().stream()
+                                     .map(exchange -> exchange.getProperty(Exchange.EXCEPTION_CAUGHT, ProcessingException.class).getCause().toString())
+                                     .map(exceptionMessage -> exceptionMessage.split("\\: ")[1])
+                                     .collect(Collectors.toList()).toArray());
+
+    }
```

Comment:
I think here you may need to run maven with the sourcechek profile and fix eventually found Code style issues.

Reply:
I have done before the PR and seems ok.
```
[INFO] --- maven-checkstyle-plugin:3.0.0:checkstyle (default-cli) @ camel-blueprint-cxf-test ---
[INFO] Starting audit...
Audit done.

```